File: infra-market-server-go/internal/dto/api_interface.go
package dto

// ApiInterfaceDto 接口信息DTO
type ApiInterfaceDto struct {
	ID           *uint64       `json:"id"`
	Name         *string       `json:"name"`
	Method       *string       `json:"method"`
	URL          *string       `json:"url"`
	Description  *string       `json:"description"`
	Status       *int          `json:"status"`
	PostType     *string       `json:"postType"`
	Environment  *string       `json:"environment"`
	Timeout      *int64        `json:"timeout"`
	ValuePath    *string       `json:"valuePath"`
	URLParams    []ApiParamDto `json:"urlParams"`
	HeaderParams []ApiParamDto `json:"headerParams"`
	BodyParams   []ApiParamDto `json:"bodyParams"`
	CreateTime   *string       `json:"createTime"`
	UpdateTime   *string       `json:"updateTime"`
}

// ApiInterfaceFormDto 接口创建/更新表单
type ApiInterfaceFormDto struct {
	ID           *uint64       `json:"id"`
	Name         *string       `json:"name" binding:"required"`
	Method       *string       `json:"method" binding:"required"`
	URL          *string       `json:"url" binding:"required"`
	Description  *string       `json:"description"`
	PostType     *string       `json:"postType"`
	Environment  *string       `json:"environment"`
	Timeout      *int64        `json:"timeout"`
	ValuePath    *string       `json:"valuePath"`
	URLParams    []ApiParamDto `json:"urlParams"`
	HeaderParams []ApiParamDto `json:"headerParams"`
	BodyParams   []ApiParamDto `json:"bodyParams"`
}

// ApiInterfaceQueryDto 接口查询DTO
type ApiInterfaceQueryDto struct {
	Name        *string `form:"name"`
	Method      *string `form:"method"`
	Status      *int    `form:"status"`
	Environment *string `form:"environment"`
	Pagination
}

// ApiParamDto 接口参数DTO
type ApiParamDto struct {
	Name         *string           `json:"name"`
	ChineseName  *string           `json:"chineseName"`
	ParamType    *string           `json:"paramType"`
	InputType    *string           `json:"inputType"`
	DataType     *string           `json:"dataType"`
	Required     *bool             `json:"required"`
	DefaultValue any               `json:"defaultValue"`
	Changeable   *bool             `json:"changeable"`
	Options      []SelectOptionDto `json:"options"`
	Description  *string           `json:"description"`
	Sort         *int              `json:"sort"`
}

// SelectOptionDto 下拉选项DTO
type SelectOptionDto struct {
	Value *string `json:"value"`
	Label *string `json:"label"`
}

// ApiExecuteRequestDto 接口执行请求DTO
type ApiExecuteRequestDto struct {
	InterfaceID *uint64           `json:"interfaceId" binding:"required"`
	Headers     map[string]string `json:"headers"`
	URLParams   map[string]any    `json:"urlParams"`
	BodyParams  map[string]any    `json:"bodyParams"`
	Timeout     *int64            `json:"timeout"`
	Remark      *string           `json:"remark"`
}

// ApiExecuteResponseDto 接口执行响应DTO
type ApiExecuteResponseDto struct {
	Status         int               `json:"status"`
	Headers        map[string]string `json:"headers"`
	Body           *string           `json:"body"`
	ExtractedValue *string           `json:"extractedValue"`
	ResponseTime   int64             `json:"responseTime"`
	Success        bool              `json:"success"`
	Error          *string           `json:"error"`
}

File: infra-market-server-dotnet/DTOs/ApiInterface.cs
using System.ComponentModel.DataAnnotations;

namespace InfraMarket.Server.DTOs;

public class ApiInterfaceDto
{
    public ulong? Id { get; set; }
    public string? Name { get; set; }
    public string? Method { get; set; }
    public string? Url { get; set; }
    public string? Description { get; set; }
    public int? Status { get; set; }
    public string? PostType { get; set; }
    public string? Environment { get; set; }
    public long? Timeout { get; set; }
    public string? ValuePath { get; set; }
    public List<ApiParamDto> UrlParams { get; set; } = [];
    public List<ApiParamDto> HeaderParams { get; set; } = [];
    public List<ApiParamDto> BodyParams { get; set; } = [];
    public string? CreateTime { get; set; }
    public string? UpdateTime { get; set; }
}

public class ApiInterfaceFormDto
{
    public ulong? Id { get; set; }

    [Required]
    public string? Name { get; set; }

    [Required]
    public string? Method { get; set; }

    [Required]
    public string? Url { get; set; }

    public string? Description { get; set; }
    public string? PostType { get; set; }
    public string? Environment { get; set; }
    public long? Timeout { get; set; }
    public string? ValuePath { get; set; }
    public List<ApiParamDto> UrlParams { get; set; } = [];
    public List<ApiParamDto> HeaderParams { get; set; } = [];
    public List<ApiParamDto> BodyParams { get; set; } = [];
}

public class ApiInterfaceQueryDto
{
    public string? Name { get; set; }
    public string? Method { get; set; }
    public int? Status { get; set; }
    public string? Environment { get; set; }
    public int? Page { get; set; }
    public int? Size { get; set; }
}

public class ApiParamDto
{
    public string? Name { get; set; }
    public string? ChineseName { get; set; }
    public string? ParamType { get; set; }
    public string? InputType { get; set; }
    public string? DataType { get; set; }
    public bool? Required { get; set; }
    public object? DefaultValue { get; set; }
    public bool? Changeable { get; set; }
    public List<SelectOptionDto> Options { get; set; } = [];
    public string? Description { get; set; }
    public int? Sort { get; set; }
}

public class SelectOptionDto
{
    public string? Value { get; set; }
    public string? Label { get; set; }
}

public class ApiExecuteRequestDto
{
    [Required]
    public ulong? InterfaceId { get; set; }
    public Dictionary<string, string>? Headers { get; set; }
    public Dictionary<string, object>? UrlParams { get; set; }
    public Dictionary<string, object>? BodyParams { get; set; }
    public long? Timeout { get; set; }
    public string? Remark { get; set; }
}

public class ApiExecuteResponseDto
{
    public int Status { get; set; }
    public Dictionary<string, string>? Headers { get; set; }
    public string? Body { get; set; }
    public string? ExtractedValue { get; set; }
    public long ResponseTime { get; set; }
    public bool Success { get; set; }
    public string? Error { get; set; }
}
